steering_controllers_library/doc/userdoc.rst

```diff
@@ -0,0 +1,88 @@
+.. _steering_controllers_libaray_userdoc:
+
+steering_controllers_library
+=============================
+
+Library with shared functionalities for mobile robots controllers with steering drive (2 degrees of freedom).
+The library implements generic odometry and update methods and defines the main interfaces.
+
+Nomenclature used for the controller is used from `wikipedia <https://en.wikipedia.org/wiki/Wheelbase>`_.
+
+Execution logic of the controller
+----------------------------------
+
+The controller uses velocity input, i.e., stamped or unstamped Twist messages where linear ``x`` and angular ``z`` components are used.
+Angular component is under
+Values in other components are ignored.
+In the chain mode the controller provides two reference interfaces, one for linear velocity and one for steering angle position.
+Other relevant features are:
+
+  - support for front and rear steering configurations;
+  - odometry publishing as Odometry and TF message;
+  - input command timeout based on a parameter.
+
+The command for the wheels are calculated using ``odometry`` library where based on concrete kinematics traction and steering commands are calculated.
+Currently implemented kinematics in corresponding packages are:
+
+  - :ref:`Bicycle <bicycle_steering_controller_userdoc>` - with one steering and one drive joints;
+  - :ref:`Tricylce <tricycle_steering_controller_userdoc>` - with one steering and two drive joints;
+  - :ref:`Ackermann <ackermann_steering_controller_userdoc>` - with two seering and two drive joints.
+
+
+
+Description of controller's interfaces
+--------------------------------------
+
+References (from a preceding controller)
+,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
+- <controller_name>/linear/velocity      [double]  # in [m/s]
+- <controller_name>/angular/position     [double]  # in [rad]
```

Comment:
As I understand the library, in a simple single-controller setup the twist (linear+angular velocity) inputs are calculated to traction wheel velocity and steering axis angle via inverse kinematics. Why do we have a position interface here? Of what type is the preceding controller in this case? Maybe you can add a paragraph on these basics, even if it is common-sense for roboticists ;) (Honestly, I never worked with these type of robots.)

Reply:
What is you are controlling the position of your joints. That the point. you can control position or velocity of specific joint. This setup enables all combinations.

Reply:
Sorry, I still do not understand it 😕 But if you sure about it, I believe you :wink: